includes/admin/legacy/class-wc-amazon-payments-advanced-order-admin-legacy.php

```diff
@@ -108,9 +110,15 @@ public function do_order_action( $order, $id, $action, $version ) {
 	 * @param int $order_id Order ID.
 	 */
 	private function clear_stored_states( $order_id ) {
-		delete_post_meta( $order_id, 'amazon_reference_state' );
-		delete_post_meta( $order_id, 'amazon_capture_state' );
-		delete_post_meta( $order_id, 'amazon_authorization_state' );
+		$order = wc_get_order( $order_id );
+		if ( ! ( $order instanceof WC_Order ) ) {
+			return;
+		}
+
+		$order->delete_meta_data( 'amazon_reference_state' );
+		$order->delete_meta_data( 'amazon_capture_state' );
+		$order->delete_meta_data( 'amazon_authorization_state' );
+		$order->save();
```

Comment:
Nitpick: can we add a new line after save, please?

Reply:
done!